### PR TITLE
fix missing title attribute on <iframe> tag suggested for embedding

### PR DESCRIPTION
**Pull request by pthavarasa**
fixed this issue. Now there is a title in the iframe tag.
share video -> video title
share playlist -> playlist title
share playlist video(right click in the video to copy embed) -> video title

### CODE REVIEW

File: client/src/assets/player/utils.ts [hunk 1]
```diff
@@ -170,9 +170,10 @@ function secondsToTime (seconds: number, full = false, symbol?: string) {
   return time
 }
 
-function buildVideoOrPlaylistEmbed (embedUrl: string) {
+function buildVideoOrPlaylistEmbed (embedUrl: string, embedTitle: string) {
```
Comment (Chocobozzz) [inline client/src/assets/player/utils.ts:173]: I think you need to escape the `embedTitle`. You can use the `escapeHTML` from `server/helpers/core-utils` after moving the function in `shared/core-utils/renderer.html.ts`

File: server/controllers/services.ts [hunk 2]
```diff
@@ -79,6 +79,7 @@ function buildOEmbed (options: {
   const embedUrl = webserverUrl + embedPath
   let embedWidth = EMBED_SIZE.width
   let embedHeight = EMBED_SIZE.height
+  let embedtitle = title
```
Comment (Chocobozzz) [inline server/controllers/services.ts:82]: Use `const` and escape it

Also it should be `embedTitle`

**Review by Chocobozzz — CHANGES_REQUESTED** [review]

File: client/src/assets/player/utils.ts (hunk 1 above)
Comment (pthavarasa) [reply client/src/assets/player/utils.ts:174]: hi @Chocobozzz can you check if i made correct changes ?

File: server/controllers/services.ts [hunk 3]
```diff
@@ -79,6 +79,7 @@ function buildOEmbed (options: {
   const embedUrl = webserverUrl + embedPath
   let embedWidth = EMBED_SIZE.width
   let embedHeight = EMBED_SIZE.height
+  const embedTitle = title
```
Comment (Chocobozzz) [inline server/controllers/services.ts:82]: Please `escapeHTML` here too

File: server/tests/api/server/services.ts [hunk 4]
```diff
@@ -88,7 +88,7 @@ describe('Test services', function () {
 
     const res = await getOEmbed(server.url, oembedUrl)
     const expectedHtml = '<iframe width="560" height="315" sandbox="allow-same-origin allow-scripts" ' +
-      `src="http://localhost:${server.port}/video-playlists/embed/${playlistUUID}" ` +
+      `title="${video.name}" src="http://localhost:${server.port}/video-playlists/embed/${playlistUUID}" ` +
```
Comment (Chocobozzz) [inline server/tests/api/server/services.ts:91]: Should be `title="The Life and Times of Scrooge McDuck" since it's the playlist embed

Comment (pthavarasa) [reply server/tests/api/server/services.ts]: can i assign directly the string ?

**Comment by Chocobozzz** [comment]
Thanks @pthavarasa!

**Comment by lunar-debian** [comment]
:+1:  Thanks @pthavarasa and @Chocobozzz !

**Pull request merged**